framework/src/base/ComputeResidualThread.C

```diff
@@ -103,6 +103,17 @@ ComputeResidualThread::onElement(const Elem *elem)
   case Moose::KT_NONTIME:
     warehouse = &_non_time_kernels;
     break;
+
+  case Moose::KT_EIGEN:
+    warehouse = &_nl.getEigenKernelWarehouse();
```

Comment:
I am not going to change the old ones. Actually, it is not necessary to store a member for this because it is used once only. That is, we access this warehouse at line 108, and then we do not access again.